### PR TITLE
feat: configurable template strings for interpolating variables for identificatie generation

### PR DESCRIPTION
**Pull request by RayetSena**

### CODE REVIEW

File: src/main/configurations/Translate/DeploymentSpecifics.properties [hunk 1]
```diff
@@ -96,6 +96,8 @@ ISO639_Codes_path=${configurations.directory}/Translate/ISO639.xml
 creeerZaak.timeout=20000
 
 Prefixparam=1900
+PrefixparamDate=1900ZK
+NewIdDigit=00000
```
Comment (MLenterman) [inline src/main/configurations/Translate/DeploymentSpecifics.properties:100]: These variables are supposed to be calculated runtime. Prefixparam is no longer needed as it is part of the template string. The date should be gotten at runtime and should be current date/time the moment the new identifier is created.
NewIdDigit should simply be the new identifier with the formatting applied to it. So the {NewIdentificatie} variable used to update the database, can be used as input for the formatting xslt.

In the properties there should only be a template string like:
zaak-identificatie-template="ZK{datetime:yyyy}-{id:D5}"

**Review by MLenterman — CHANGES_REQUESTED** [review]

File: src/main/configurations/Translate/Common/xsl/CreateIdentication.xslt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            <xsl:for-each
+            select="$patternArray">
+            <xsl:choose>
+                <xsl:when test="contains(current(), 'datetime')">
+                    <xsl:value-of select="substring-before(current(),'{')" />
+                        <xsl:variable
+                        name="date"
+                        select="substring-before(substring-after(current(),':'), '}')" />
+                        <xsl:value-of
+                        select="concat(format-dateTime(current-dateTime(), $date),'-')" />
```
Comment (MLenterman) [inline src/main/configurations/Translate/Common/xsl/CreateIdentication.xslt:21]: Is missing a check for the optional ':', indicating that a format string follow. Without ':' it should just return current-dateTime without any formatting.

File: src/main/configurations/Translate/DeploymentSpecifics.properties [hunk 3]
```diff
@@ -95,7 +95,7 @@ ISO639_Codes_path=${configurations.directory}/Translate/ISO639.xml
 
 creeerZaak.timeout=20000
 
-Prefixparam=1900
+zaak-identificatie-template=ZK{datetime:[Y0001]}-{id:D5}
```
Comment (MLenterman) [inline src/main/configurations/Translate/DeploymentSpecifics.properties:98]: The same mechanism should also be implemented for document identificatie with its own template in the DeploymentSpecifics.

File: src/main/configurations/Translate/Configuration_GenereerIdentificatieEmulator.xml [hunk 4]
```diff
@@ -43,9 +43,13 @@
 			<IncreaseIntegerPipe 
                 name="IncrementNewIdentificatie"
                 sessionKey="NewIdentificatie">
-                <Forward name="success" path="UpdateIdentificatie" />
+                <Forward name="success" path="CreateNewId" />
             </IncreaseIntegerPipe>
 
+            <XsltPipe name="CreateNewId" styleSheetName="Common/xsl/CreateIdentication.xslt" storeResultInSessionKey="newId">
+                <Param name="zaak-identificatie-template" value="${zaak-identificatie-template}"/>
```
Comment (MLenterman) [inline src/main/configurations/Translate/Configuration_GenereerIdentificatieEmulator.xml:50]: Zaak en document both have their own template string in the DeploymentSpecifics. Probably pass the template as session key to the adapter. While calling it from the genereerZaakIdentificatie and genereerDocumentIdentificatie adapters you can pass the correct one.

**Review by MLenterman — CHANGES_REQUESTED** [review]

**Review by philipsens — APPROVED** [review]

**Review by MLenterman — APPROVED** [review]

**Pull request merged**